src/main/java/core/basesyntax/impl/AmountFromFileImpl.java

```diff
@@ -0,0 +1,42 @@
+package core.basesyntax.impl;
+
+import core.basesyntax.db.FruitShopDao;
+import core.basesyntax.service.FruitTransaction;
+import core.basesyntax.service.AmountFromFile;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class AmountFromFileImpl implements AmountFromFile {
+    private FruitShopDao fruitShopDao;
+
+    public AmountFromFileImpl(FruitShopDao fruitShopDao) {
+        this.fruitShopDao = fruitShopDao;
+    }
+
+    @Override
+    public List<Integer> getAmountInShop(List<String> fruits,
+                                         List<FruitTransaction> fruitTransactions) {
+        List<Integer> amounts = new ArrayList<>();
+
+        for (int i = 0; i < fruits.size(); i++) {
+            String fruit = fruits.get(i);
+            int sum = fruitTransactions.stream()
+                    .filter(ft -> ft.getFruit().equals(fruit))
+                    .mapToInt(n -> calculateQantityFromOperation(n))
+                    .sum();
+            amounts.add(sum);
+            fruitShopDao.put(fruit,sum);
```

Comment:
Directly updating the `fruitShopDao` within the loop violates the Single Responsibility Principle. This method should only be responsible for calculating the amounts, not for updating the storage. Consider removing this line and handling the storage update in a separate method or class.